osfoffline/polling_osf_manager/polling_events.py

```diff
@@ -21,117 +21,150 @@ def run(self):
 
 class CreateFolder(PollingEvent):
     def __init__(self, path):
-        super().__init__(path)
-        self.path = ProperPath(path, is_dir=True)
-        assert self.path
+        super().__init__()
+        self.path = path
 
     @asyncio.coroutine
     def run(self):
         # create local node folder on filesystem
-        if not os.path.exists(self.path.full_path):
-            AlertHandler.info(self.path.name, AlertHandler.DOWNLOAD)
-            os.makedirs(self.path.full_path)
+        try:
+            folder_to_create = ProperPath(self.path, is_dir=True)
+        except Exception:
+            # TODO: Narrow down this exception and do client side warnings
+            logging.exception('Exception caught: Invalid target path for folder.')
+            return
+        if not os.path.exists(folder_to_create.full_path):
```

Comment:
True. I was trying not to otherwise affect functionality with this, only ensure that events are handled, but I can make changes if you'd like.